paddle/operators/recv_op.cc

```diff
@@ -95,27 +93,33 @@ class RecvOp : public framework::OperatorBase {
     auto param_list = Attr<std::vector<std::string>>("ParamList");
     auto grad_list = Attr<std::vector<std::string>>("GradList");
     auto fan_in = Attr<int>("Fanin");
-    size_t param_count = param_list.size();
 
     auto *block = Attr<framework::BlockDesc *>(kOptimizeBlock);
     auto *program = block->Program();
     framework::Executor executor(dev_place);
 
     // TODO(typhoonzero): change this to a while_op for every cluster-batch.
     bool exit_flag = false;
-    size_t barrier_size = param_count * fan_in;
     while (!exit_flag) {
       // Get from multiple trainers, we don't care about the order in which
       // the gradients arrives, just add suffix 0~n and merge the gradient.
       rpc_service_->SetCond(0);
-      for (size_t i = 0; i < barrier_size; ++i) {
+      size_t barrier_size = 0;
+      int batch_barrier = 0;
+      while (batch_barrier != fan_in || !rpc_service_->IsRecvQueueEmpty()) {
```

Comment:
Not sure why `!rpc_service_->IsRecvQueueEmpty()` is needed. `rpc_service_->Get()` is a blocking call which will wait until a new message arrives.

Reply:
Thanks, it's not used :)

Reply:
I deleted `!rpc_service_->IsRecvQueueEmpty()`, because send op would send barrier signal by least, if RecvOp received barrier signal, it should be the least message from one trainer.